internal/pkg/service/common/etcdop/mutex.go

```diff
@@ -292,9 +294,9 @@ func (m *activeMutex) localUnlock() error {
 func (m *activeMutex) dbLock(ctx context.Context) (err error) {
 	// Get session at the first time
 	if m.dbSession == nil {
-		m.dbSession, err = m.store.session.Session()
+		m.dbSession, err = m.store.session.WaitForSession(ctx)
 		if err != nil {
-			return err
+			return errors.PrefixError(err, "cannot get concurrency session")
 		}
```

Comment:
We should wait for the session, until the context is cancelled.

It is edge-case, when the connection is broken - I noticed it in CI, because there are fewer resources, and etcd is sometimes overloaded with parallel tests - it doesn't respond for a while.
​